src/System.Net.Security/src/System/Net/Unix/CertModule.cs

```diff
@@ -76,6 +76,33 @@ internal override X509Certificate2 GetRemoteCertificate(SafeDeleteContext securi
                     remoteContext.DangerousAddRef(ref gotReference);
                     result = new X509Certificate2(remoteContext.DangerousGetHandle());
                 }
+
+                remoteCertificateStore = new X509Certificate2Collection();
+
+                SafeSharedX509StackHandle chainStack =
+                    Interop.OpenSsl.GetPeerCertificateChain(securityContext.DangerousGetHandle());
```

Comment:
GC.KeepAlive is sufficient. If https://github.com/dotnet/corefx/pull/3528 goes after this, I will fix this up